src/axom/spin/tests/spin_bvh.cpp

```diff
@@ -801,23 +1350,43 @@ AXOM_CUDA_TEST( spin_bvh, contruct3D_cuda )
 }
 
 //------------------------------------------------------------------------------
-AXOM_CUDA_TEST( spin_bvh, find_3d_cuda )
+TEST( spin_bvh, find_rays_3d_cuda )
+{
+  constexpr int BLOCK_SIZE = 256;
+  using exec  = axom::CUDA_EXEC< BLOCK_SIZE >;
+
+  check_find_rays3d< exec, float >( );
+  check_find_rays3d< exec, double >( );
+}
+
+//------------------------------------------------------------------------------
+TEST( spin_bvh, find_rays_2d_cuda )
+{
+  constexpr int BLOCK_SIZE = 256;
+  using exec  = axom::CUDA_EXEC< BLOCK_SIZE >;
+
+  check_find_rays2d< exec, float >( );
+  check_find_rays2d< exec, double >( );
+}
+
+//------------------------------------------------------------------------------
+AXOM_CUDA_TEST( spin_bvh, find_points_3d_cuda )
 {
   constexpr int BLOCK_SIZE = 256;
   using exec  = axom::CUDA_EXEC< BLOCK_SIZE >;
 
-  check_find3d< exec, float >( );
-  check_find3d< exec, double >( );
+  check_find_points3d< exec, float >( );
```

Comment:
@mclarsen, @rhornung67 for some reason my tests are failing on the GPU when using  single precision, i.e.,`float`. They are passing with `double`.  

Any ideas? I saw some discussion about the RAJA scans no working properly with single precision. Could this be related to that?

Reply:
Are you using RAJA scan operations?  These may not work properly in parallel (CUDA or OpenMP) due to low precision and/or non-associativity of plus operations.

Reply:
Yes, we are using scan operations. They do not work properly with single precision, but, they work properly when using double precision. Correct?

Reply:
Correct. Not sure how to fix this. We are using CUB internally in the scans.

Reply:
Thanks @rhornung67 , I'll just remove the tests with single precision -- I guess we shouldn't be thoroughly testing things :)

Reply:
Well, I just learned about this issue recently as I was beefing up the RAJA scan tests. So at least we know about it now.

Reply:
Yeah, I saw Will's comment. That's what prompted me to see if the tests are passing with just double precision. 

@mclarsen you may in particular want to take note of that since I think you are intrinsically using single precision in your code as I recall...

Reply:
@mclarsen do you have a solution for parallel scans that works for float data?